src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/DivInstruction.cs

```diff
@@ -172,202 +173,19 @@ public override int Run(InterpretedFrame frame)
 
         public static Instruction Create(Type type)
         {
-            Debug.Assert(!type.GetTypeInfo().IsEnum);
+            Debug.Assert(type.IsArithmetic());
             switch (type.GetNonNullableType().GetTypeCode())
             {
-                case TypeCode.Int16: return s_int16 ?? (s_int16 = new DivInt16());
-                case TypeCode.Int32: return s_int32 ?? (s_int32 = new DivInt32());
-                case TypeCode.Int64: return s_int64 ?? (s_int64 = new DivInt64());
+                case TypeCode.Int16: return s_Int16 ?? (s_Int16 = new DivInt16());
+                case TypeCode.Int32: return s_Int32 ?? (s_Int32 = new DivInt32());
+                case TypeCode.Int64: return s_Int64 ?? (s_Int64 = new DivInt64());
                 case TypeCode.UInt16: return s_UInt16 ?? (s_UInt16 = new DivUInt16());
                 case TypeCode.UInt32: return s_UInt32 ?? (s_UInt32 = new DivUInt32());
                 case TypeCode.UInt64: return s_UInt64 ?? (s_UInt64 = new DivUInt64());
-                case TypeCode.Single: return s_single ?? (s_single = new DivSingle());
-                case TypeCode.Double: return s_double ?? (s_double = new DivDouble());
-
-                default:
-                    throw Error.ExpressionNotSupportedForType("Div", type);
-            }
-        }
-    }
-
-    internal abstract class ModuloInstruction : Instruction
```

Comment:
Moved to separate file. To reduce the diff here, I'd like to push a separate PR (after https://github.com/dotnet/corefx/pull/14180) to split files where needed.